### DISCUSSION TITLE
Preliminary experiments phase

### DISCUSSION THREAD

**Discussion by NeuroThom** [SainsburyWellcomeCentre/PhieldsExperiments#3] [General]
Preliminary data required for applications in the first months of 2025. Ideally there will be five experiments, for which workflows will need to be built and tested:

<u>Experiment Timeline:</u>

```mermaid
%%{init: {"theme": "dark"}}%%
gantt
    title Preliminary Phields Experiments
    dateFormat  YYYY-MM-DD
    
    section Aeon3/4
    Experiment 1 - Patch Difficulty Switching    :2024-01-07, 2024-01-21
        
    section Proto-Phields
    Experiment 2/3  :2024-01-10, 2024-01-18
    Experiment 4     :2024-01-17, 2024-01-27
    Experiment 5     :2024-01-26, 2024-02-04

    section Week Markers
    Week Commencing :2024-01-06, 2024-01-06
    Week 2 :2024-01-13, 2024-01-13
    Week 3 :2024-01-20, 2024-01-20
    Week 4 :2024-01-27, 2024-01-27
```
<u>Experiment outline:</u>

1. Switching patch difficulty. 
    - On Aeon
    - Single mouse
    - CV Centroid tracking only 
    - Two patches of equal difficulty. Switch difficulty and assess switch in behaviour. Ideally switching would happened automatically like in presocial. The task switches once the mouse  have gotten enough pellets from both patches. If this doesn’t happen the task switches automatically after X minutes. The patch that become harder is the one from which the mouse got more pellets at switching time.
    - Should be doable with Aeon workflows
- 3-4 hrs experiment
    
---
LOOM NOTES! 

- Triggers should be probabilistic (each trigger has a chance to trigger a loom)
- Once a loom has been triggered there should be a refractory period during which no loom can be triggered
---

2. Multi-shelter loom escape
    - On Proto-Phields
    - Single mouse
    - Single camera covering whole arena (choose Flir, high-res or lo-res, EV camera?)
    - CV Centroid tracking only
    - 2 underground burrowable tiles as shelters
    - Only one shelter open at any one time - switches done manually for now, automated in future
    - Mechanical loom as a harp device
    - Automated triggering of the loom the loom triggers with a certain probability P, if the mouse is inside a certain ROI for X ms and is moving towards the loom within a certain angle.
- 3-4 hours

3. Occluded path escape
    - Workflow wise, is identical to experiment 2
- 3-4 hours

4. Split habitat
    - On Proto-Phields, split into two equal arenas with Aeon access gate
    - Single mouse
    - Single camera
    - Mechanical loom (placed over the chosen nest)
    - CV Centroid tracking only
    - Day 1 allow nest building, Day 2 repeated loom over nest
    - Determine if/when mouse switches side to nest
    -  Automated triggering of the loom the loom triggers with a certain probability P, if the mouse is inside a certain ROI for X ms and is moving towards the loom within a certain angle.

- 2 days. End point when mouse switches side
- Need to monitor night/day cycle with camera
 

5. Prior stress / hypervigilance
    - On Proto-Phields
    - Two mice, one stressed, one not
    - Single camera
    - Ideally avoid SLEAP identity tracking; (bleach / shave fur, tattoo, but will need testing of different computer vision strategies for ID)
    - Mechanical loom

- 3-4 hrs

**Comment by aspaNeuro** [comment]
Comments;
 - Harp H2 loom will be developed in parallel and only integrated into the experiment after it is implemented and tested. Meanwhile, we will use the current implementation.
 - Need to buy spare components to assemble a 2nd loom.
 - Experiments 3 and 4 should only start after experiment 1 running stable for 2 or 3 days.
 - Only one shelter open at any one time should have a manual input in the workflow.
 - Automated triggering:
    -  Needs testing before is applied to experiments (button will always work )
    - The mouse is inside a certain ROI for X ms 
    - And is moving towards the loom within a certain angle (velocity ??)
    - All of these features don't emerge from simple blob tracking so custom development and reuse of other projects solutions need to be applied.
 - For the AEON solution the task control could be simplified to match the requirements:
    -  A possible solution is to use the trigger mapping to allow multiple independent components to create state transitions
    - Also yaml to describe the task, we need more info on:
    `Two patches of equal difficulty. Switch difficulty and assess switch in behaviour. Ideally switching would happened automatically like in presocial. The task switches once the mouse have gotten enough pellets from both patches. If this doesn’t happen the task switches automatically after X minutes. The patch that become harder is the one from which the mouse got more pellets at switching time.`
    - what does harder or easier means in this context, is it the same or  we can create a sequence of easy and harder parameters to be applied after each transition, etc.
 - There is a need to define priorities between AEON and PHIELDS in the usage of AEON arenas.
 - Regarding mouse ID, is there a need to do online id tracking for the task, according to task descriptions there is none: `Ideally avoid SLEAP identity tracking; (bleach / shave fur, tattoo, but will need testing of different computer vision strategies for ID)`
 - While preparing the experiments Emergent Tech developments and integrations need to be on hold, so we have time to follow priorities.

**Comment by Dario55** [comment]
Comments;

Harp H2 loom will be developed in parallel and only integrated into the experiment after it is implemented and tested. Meanwhile, we will use the current implementation.
Need to buy spare components to assemble a 2nd loom.

ABOVE 2 POINTS NOT REQUIRED FOR THE PILOT.

Experiments 3 and 4 should only start after experiment 1 running stable for 2 or 3 days.
DIDN’T UNDERSTAND IT. EXPERIMENT ONE IS THE AEON ARENA RIGHT?

Only one shelter open at any one time should have a manual input in the workflow.

Automated triggering:
Needs testing before is applied to experiments (button will always work )
YES I AGREE, LETS ALWAYS KEEP THE OPTION  FROM CTRL+D IN CASE THINGS DONT WORK. 

The mouse is inside a certain ROI for X ms
And is moving towards the loom within a certain angle (velocity ??)
GOOD POINT ALSO A RANGE IN VELOCITY CAN BE A USEFUL FEATURE TO HAVE.
All of these features don't emerge from simple blob tracking so custom development and reuse of other projects solutions need to be applied.


For the AEON solution the task control could be simplified to match the requirements:
A possible solution is to use the trigger mapping to allow multiple independent components to create state transitions
Also yaml to describe the task, we need more info on:
Two patches of equal difficulty. Switch difficulty and assess switch in behaviour. Ideally switching would happened automatically like in presocial. The task switches once the mouse have gotten enough pellets from both patches. If this doesn’t happen the task switches automatically after X minutes. The patch that become harder is the one from which the mouse got more pellets at switching time.
what does harder or easier means in this context, is it the same or we can create a sequence of easy and harder parameters to be applied after each transition, etc.

EASY AND HARD CAN BE DEFINED IN THE SAME WAY AS WE ARE DOING FOR SOCIAL. WE START WITH SOMETHING THAT GIVES AROUND ONE PELLET PER METER IN BOTH PATCHES AND THEN ONE JUMPS TO LIKE 10 METER AND THE OTHER STAYS THE SAME.

There is a need to define priorities between AEON and PHIELDS in the usage of AEON arenas.
HAVE ALREADY SPOKEN TO JAI. THE MAIN ARENA USAGE FOR JANUARY WAS THE SOCIAL EPHYS BUT HE IS HAPPY TO PUSH IT OF A FEW WEEKS FORWARD. HE MAY ALSO
NOT BE AROUND MUCH KN JANUARY DUE TO AN INTERSHIP ANYWAY.

Regarding mouse ID, is there a need to do online id tracking for the task, according to task descriptions there is none: Ideally avoid SLEAP identity tracking; (bleach / shave fur, tattoo, but will need testing of different computer vision strategies for ID)
YES IT CAN ALWAYS BE DONE POSTHOC, DUE
TO THE RUSHED NATURE OF THE PILOT THE MAIN ANDVANTAGE OF GETTING IT WORK ONLINE IS TO SAVE TIME WITH RETRAKING, BUT WE HAVE TO BE REALISTIC. 

While preparing the experiments Emergent Tech developments and integrations need to be on hold, so we have time to follow priorities.
YEP MAKES SENSE, HAVE WRITTEN A COMMENT ON THIS IN TEAMS

**Comment by Dario55** [comment]
![Image](https://github.com/user-attachments/assets/00fdd833-985d-45ad-b9ac-8f9683d75ada)
NOTE.
It might be that we would invert experiment 4 and experiment 2/3, but the workflow requirements would be the same. 
Another thing to consider is that while the Phileds arena will be heavily used, the foraging one is always free for the period, so I would suggest to make sure that Phileds is ready first so we can start straight away, while for aeon we could start 2-3 days later

**Reply by Dario55** [reply]
IGNORE THIS GANTT

**Comment by Dario55** [comment]
I have edited the experiment timeline and anticipated the start of the first proto-Phileds experiment to the 9th. It is only one day but it could make the difference given the short time frame available. In practice the decision of when we start it is also dependent on when @lorycalca is back. If she is already around on the 2nd then we could plan some testing, hardware preparation and maybe food deprivation for the 2nd, 3rd and 6th and maybe start straight on the 7th. 

I have reduced the number of shelters in the multi-shelters experiments to 2, it is going to be sufficient for what we need and will make experiment, analysis and possibly the workflow development simpler.

**Comment by Dario55** [comment]
Schematic of experiments in room 462: 
![image](https://github.com/user-attachments/assets/7bcb3f00-0e80-4bdb-9e30-07beb53d0cac)
![image](https://github.com/user-attachments/assets/c6875482-5bc4-47f9-bb2c-d86268baf274)
![image](https://github.com/user-attachments/assets/a3102e09-56ed-4fcf-8f21-960d6eab5341)
